subsys/bluetooth/mesh/Kconfig

```
@@ -895,6 +895,80 @@ config BT_MESH_BLOB_REPORT_TIMEOUT
 	help
 	  The timer value that Pull BLOB Transfer Server uses to report missed chunks.
 
+config BT_MESH_RX_BLOB_CHUNK_SIZE
+	int "BLOB Server chunk size"
+	default 377 if BT_MESH_RX_SEG_MAX = 32
+	default 365 if BT_MESH_RX_SEG_MAX = 31
+	default 353 if BT_MESH_RX_SEG_MAX = 30
+	default 341 if BT_MESH_RX_SEG_MAX = 29
+	default 329 if BT_MESH_RX_SEG_MAX = 28
+	default 317 if BT_MESH_RX_SEG_MAX = 27
+	default 305 if BT_MESH_RX_SEG_MAX = 26
+	default 293 if BT_MESH_RX_SEG_MAX = 25
+	default 281 if BT_MESH_RX_SEG_MAX = 24
+	default 269 if BT_MESH_RX_SEG_MAX = 23
+	default 257 if BT_MESH_RX_SEG_MAX = 22
+	default 245 if BT_MESH_RX_SEG_MAX = 21
+	default 233 if BT_MESH_RX_SEG_MAX = 20
+	default 221 if BT_MESH_RX_SEG_MAX = 19
+	default 209 if BT_MESH_RX_SEG_MAX = 18
+	default 197 if BT_MESH_RX_SEG_MAX = 17
+	default 185 if BT_MESH_RX_SEG_MAX = 16
+	default 173 if BT_MESH_RX_SEG_MAX = 15
+	default 161 if BT_MESH_RX_SEG_MAX = 14
+	default 149 if BT_MESH_RX_SEG_MAX = 13
+	default 137 if BT_MESH_RX_SEG_MAX = 12
+	default 125 if BT_MESH_RX_SEG_MAX = 11
+	default 113 if BT_MESH_RX_SEG_MAX = 10
+	default 101 if BT_MESH_RX_SEG_MAX = 9
+	default 89 if BT_MESH_RX_SEG_MAX = 8
+	default 77 if BT_MESH_RX_SEG_MAX = 7
+	default 65 if BT_MESH_RX_SEG_MAX = 6
+	default 53 if BT_MESH_RX_SEG_MAX = 5
+	default 41 if BT_MESH_RX_SEG_MAX = 4
+	default 29 if BT_MESH_RX_SEG_MAX = 3
+	default 17 if BT_MESH_RX_SEG_MAX = 2
+	default 8 # unsegmented
```

Considering mesh v1.1 application will be compiled with maximum payload length for use other than DFU, I suggest, we can remove this, and keep it only as "default 8". This is the value that will always work without any compilation errors.
```suggestion	
	default 8 # unsegmented
```

And in the DFU target sample, we override this and limit it to "64" (while setting TX/RX segments to 32). Reason for choosing 64 is, to use some number that will result into PDU segments higher than "SAR Segments threshold", but not too high to cause many segments.

---

Not sure if I agree. From a view of user having the defaults automatically set to max possible might be favourable (and some applications from customers might break then as we change default behaviour)
(Plus, FYI, I'll have to modify the prf.conf file for bsim tests then to increase as well (either max, or also 64, or something in that range) if it is lower there is a whole lot of changes to be done so dfu doesn't time out during the friendship tests, etc. etc.)

---

It is fine to not change this for me. Though I feel the issue that you mentioned is not as big as you might be expecting.

---

Well, we want the user to not be able to select a value that is not in range, nor should the default behaviour change. Plus from user perspective, all these ifs shouldn't be visible, right?
(And the point of this change is to give the user an option to change the chunk size)

---

If customer wants to customize value, it is up to him\her to follow up value correctness.
The default behavior that adapts chunk size to segment size does it. This is sufficient.